addons/web/static/src/js/views/sample_server.js

```diff
@@ -622,14 +633,22 @@
             this._populateExistingGroups(params);
 
             // update count and aggregates for each group
-            const groupBy = params.groupBy[0].split(':')[0];
-            const groupByField = this.data[params.model].fields[groupBy];
-            const groupedByM2O = groupByField.type === 'many2one';
+            const groupBy = params.groupBy[0];
+            // since date/datetime groupBy values could have a granularity, we have to filter it out
+            const groupByFieldName = groupBy.split(':')[0];
+            const groupByField = this.data[params.model].fields[groupByFieldName];
             const records = this.data[params.model].records;
             for (const g of groups) {
-                const groupValue = groupedByM2O ? g[groupBy][0] : g[groupBy];
-                const recordsInGroup = records.filter(r => r[groupBy] === groupValue);
-                g[`${groupBy}_count`] = recordsInGroup.length;
+                var groupValue = g[groupBy];
+                if (groupByField.type === 'many2one') {
+                    groupValue = g[groupBy][0];
+                } else if (['date', 'datetime'].includes(groupByField.type)) {
```

Comment:
Could tell in commit message that we add support of datetime-based grouping in sample view.